examples/Classification/pointcloud_classification.py

```diff
@@ -16,28 +17,43 @@
 parser.add_argument('--epochs', type=int, default=10, help='Number of train epochs.')
 parser.add_argument('-lr', '--learning-rate', type=float, default=1e-3, help='Learning rate.')
 parser.add_argument('--batch-size', type=int, default=12, help='Batch size.')
-parser.add_argument('--device', type=str, default='cuda', help='Device to use.')
+parser.add_argument('--viz-test', action='store_true', help='Visualize an output of a test sample')
+parser.add_argument('--transforms-device', type=str, default='cuda', help='Device to use.')
 
 args = parser.parse_args()
 
 
+def to_device(inp):
+    inp.to(args.transforms_device)
+    return inp
+
 transform = tfs.Compose([
+    to_device,
     tfs.TriangleMeshToPointCloud(num_samples=args.num_points),
     tfs.NormalizePointCloud()
 ])
 
+if args.transforms_device == 'cuda':
+    num_workers = 0
+    pin_memory = False
+else:
+    num_workers = cpu_count()
+    pin_memory = True
+
 train_loader = DataLoader(ModelNet(args.modelnet_root, categories=args.categories,
-                                   split='train', transform=transform, device=args.device),
-                          batch_size=args.batch_size, shuffle=True)
+                                   split='train', transform=transform),
+                          batch_size=args.batch_size, shuffle=True,
+                          num_workers=num_workers, pin_memory=pin_memory)
 
 val_loader = DataLoader(ModelNet(args.modelnet_root, categories=args.categories,
-                                 split='test', transform=transform, device=args.device),
-                        batch_size=args.batch_size)
+                                 split='test', transform=transform),
+                        batch_size=args.batch_size,
+                        num_workers=num_workers, pin_memory=pin_memory)
 
-model = PointNetClassifier(num_classes=len(args.categories)).to(args.device)
+model = PointNetClassifier(num_classes=len(args.categories)).to('cuda')
 optimizer = torch.optim.Adam(model.parameters(), lr=args.learning_rate)
 criterion = torch.nn.CrossEntropyLoss()
-
+start_time = time.time()
```

Comment:
How about using `tqdm` / `trange`? We already have it as a dependency.

Downside is we might not want to litter logs with progress bars. I'm fine with either way.

Reply:
Later we'd like to grep those metrics (accuracy, training time) `tests/examples/*.sh` to check that we don't have speed regression. Not sure I can do that easily with tqdm logs.

Reply:
Agree; tqdm litters a lot